### PR TITLE
Fix Gitlab module deprecation warning

### LINKED ISSUE #59976: gitlab_runner module: Misleading deprecation warnings in Ansible 2.8
##### SUMMARY
The gitlab_runner module throws deprecation warnings for an option I don't use and another option that is not actually deprecated

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

'gitlab_runner' module

##### ANSIBLE VERSION

```
ansible 2.8.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/m/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/m/.local/lib/python3.6/site-packages/ansible
  executable location = /home/m/.local/bin/ansible
  python version = 3.6.8 (default, Jan 14 2019, 11:02:34) [GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
```

##### CONFIGURATION

```paste below
DEFAULT_REMOTE_USER(/etc/ansible/ansible.cfg) = root
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Target: Debian Buster, freshly installed. Kernel 4.19.0-5-amd64, Python versions 2.7.16 and 3.7.3

##### STEPS TO REPRODUCE

Create a task using the gitlab_runner module. Here is the task I use:

```
m@void:~/work/ansible-default-deployment$ cat roles/gitlab-runner/tasks/main.yml 
---
- name: Install python-gitlab via pip
  pip:
      name: python-gitlab
      state: present

- name: Register runner
  gitlab_runner:
      url: <redacted>
      api_token: "{{ gitlab_api_token }}"
      registration_token: "{{ gitlab_runner_registration_token }}"
      description: "Auto-created runner for project {{ project_name }}"
      state: present
      active: False
      tag_list: ['docker', '{{project_name}}']
      run_untagged: False
      locked: False
```

##### EXPECTED RESULTS

- I expect no deprecation warning for the `url` parameter because it is described as a required parameter with no possible alternative, and it is not marked as deprecated even in the 'devel' docs
- I expect no deprecation warning for the `login_token`  parameter because I don't use it

##### ACTUAL RESULTS

This is the output of running the aforenmentioned task:

```
TASK [gitlab-runner : Register runner] *************************************************************************************************************************************************************************************************************
[DEPRECATION WARNING]: Param 'url' is deprecated. See the module docs for more information. This feature will be removed in version 2.10. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: Aliases 'login_token' are deprecated. This feature will be removed in version 2.10. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
changed: [redacted]
```

### PR DESCRIPTION
**Pull request by Lunik**
##### SUMMARY

Fix and enchance deprecation warning for gitlab modules.
Also fix `required` parameters that should not be required.

Fixes #59976 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

gitlab_runner
gitlab_user
gitlab_project
gitlab_group
gitlab_hook
gitlab_deploy_key

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Lunik** [comment]
shipit

**Comment by gundalow** [comment]
So sounds like we need to merge and backport this (before 2.9.0 is released) then we can merge https://github.com/ansible/ansible/pull/61912 into devel (for release in 2.10)

I've triggered a rerun of CI. Could anyone please help review this GitLab PR, looks fairly straight forward, though review from someone that knows the area better would help. Thanks in advance.

**Comment by Shaps** [comment]
Since this has taken longer than the 2.9 release, I'd rather start removing the parameters directly so we can get it done by the time 2.10 is released!

File: lib/ansible/modules/source_control/gitlab_deploy_key.py [hunk 1]
```diff
@@ -241,7 +241,9 @@ def deleteDeployKey(self):
 def deprecation_warning(module):
     deprecated_aliases = ['private_token', 'access_token']
 
-    module.deprecate("Aliases \'{aliases}\' are deprecated".format(aliases='\', \''.join(deprecated_aliases)), "2.10")
+    for aliase in deprecated_aliases:
+        if aliase in module.params:
+            module.deprecate("Aliases \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```
Comment (Shaps) [inline lib/ansible/modules/source_control/gitlab_deploy_key.py]:
```suggestion
            module.deprecate("Alias \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```

File: lib/ansible/modules/source_control/gitlab_group.py [hunk 2]
```diff
@@ -270,7 +269,9 @@ def existsGroup(self, project_identifier):
 def deprecation_warning(module):
     deprecated_aliases = ['login_token']
 
-    module.deprecate("Aliases \'{aliases}\' are deprecated".format(aliases='\', \''.join(deprecated_aliases)), "2.10")
+    for aliase in deprecated_aliases:
+        if aliase in module.params:
+            module.deprecate("Aliases \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```
Comment (Shaps) [inline lib/ansible/modules/source_control/gitlab_group.py]:
```suggestion
            module.deprecate("Alias \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```

File: lib/ansible/modules/source_control/gitlab_hook.py [hunk 3]
```diff
@@ -297,9 +297,11 @@ def deleteHook(self):
 
 
 def deprecation_warning(module):
-    deprecated_aliases = ['private_token', 'access_token']
+    deprecated_aliases = ['private_token', 'access_token', 'enable_ssl_verification']
 
-    module.deprecate("Aliases \'{aliases}\' are deprecated".format(aliases='\', \''.join(deprecated_aliases)), "2.10")
+    for aliase in deprecated_aliases:
+        if aliase in module.params:
+            module.deprecate("Aliases \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```
Comment (Shaps) [inline lib/ansible/modules/source_control/gitlab_hook.py]:
```suggestion
            module.deprecate("Alias \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```

File: lib/ansible/modules/source_control/gitlab_project.py [hunk 4]
```diff
@@ -289,7 +288,9 @@ def existsProject(self, namespace, path):
 def deprecation_warning(module):
     deprecated_aliases = ['login_token']
 
-    module.deprecate("Aliases \'{aliases}\' are deprecated".format(aliases='\', \''.join(deprecated_aliases)), "2.10")
+    for aliase in deprecated_aliases:
+        if aliase in module.params:
+            module.deprecate("Aliases \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```
Comment (Shaps) [inline lib/ansible/modules/source_control/gitlab_project.py]:
```suggestion
            module.deprecate("Alias \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```

File: lib/ansible/modules/source_control/gitlab_runner.py [hunk 5]
```diff
@@ -286,15 +285,17 @@ def deleteRunner(self):
 
 
 def deprecation_warning(module):
-    deprecated_aliases = ['login_token']
+    deprecated_aliases = ['private_token']
 
-    module.deprecate("Aliases \'{aliases}\' are deprecated".format(aliases='\', \''.join(deprecated_aliases)), "2.10")
+    for aliase in deprecated_aliases:
+        if aliase in module.params:
+            module.deprecate("Aliases \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```
Comment (Shaps) [inline lib/ansible/modules/source_control/gitlab_runner.py]:
```suggestion
            module.deprecate("Alias \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```

File: lib/ansible/modules/source_control/gitlab_user.py [hunk 6]
```diff
@@ -411,13 +410,15 @@ def deleteUser(self):
 def deprecation_warning(module):
     deprecated_aliases = ['login_token']
 
-    module.deprecate("Aliases \'{aliases}\' are deprecated".format(aliases='\', \''.join(deprecated_aliases)), "2.10")
+    for aliase in deprecated_aliases:
+        if aliase in module.params:
+            module.deprecate("Aliases \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```
Comment (Shaps) [inline lib/ansible/modules/source_control/gitlab_user.py]:
```suggestion
            module.deprecate("Alias \'{aliase}\' is deprecated".format(aliase=aliase), "2.10")
```

**Review by Shaps — CHANGES_REQUESTED** [review]
@Lunik Since we'll be printing 1 deprecation per alias, I think it makes more sense to change `Aliases` with `Alias`. Looks good otherwise

**Comment by Lunik** [comment]
Thank's for the review @Shaps. It should be fine now

**Comment by Lunik** [comment]
ready_for_review

**Comment by Shaps** [comment]
shipit

**Pull request merged**